lib/fastlane/plugin/revenuecat_internal/helper/revenuecat_internal_helper.rb

```diff
@@ -13,12 +13,11 @@ module Fastlane
 
   module Helper
     class RevenuecatInternalHelper
-      def self.replace_version_number(previous_version_number, new_version_number, files_to_update, files_to_update_without_prerelease_modifiers)
+      def self.replace_version_number(previous_version_number, new_version_number, files_to_update_with_patterns, files_to_update_without_prerelease_modifiers)
         previous_version_number_without_prerelease_modifiers = previous_version_number.split("-")[0]
         new_version_number_without_prerelease_modifiers = new_version_number.split("-")[0]
-
-        files_to_update.each do |file_to_update|
-          replace_in(previous_version_number, new_version_number, file_to_update)
+        files_to_update_with_patterns.each do |file_to_update, pattern|
```

Comment:
There are some situations where we have to change it multiple times in the same file (don't remember which but I'm pretty sure there are. So maybe we could pass a map of file to update, to an array of patterns?

Reply:
+1

Reply:
oh I like that, I will add that